gcs_contents_manager.py

```diff
@@ -428,3 +449,262 @@ def rename_file(self, old_path, new_path):
       raise err
     except Exception as ex:
       raise HTTPError(500, 'Internal server error: {}'.format(str(ex)))
+
+
+class CombinedCheckpointsManager(GenericCheckpointsMixin, Checkpoints):
+
+  def __init__(self, content_managers):
+    self._content_managers = content_managers
+
+  def _checkpoint_manager_for_path(self, path):
+    path = path or ''
+    path = path.strip('/')
+    for path_prefix in self._content_managers:
+      if path == path_prefix or path.startswith(path_prefix+'/'):
+        relative_path = path[len(path_prefix):]
+        return self._content_managers[path_prefix].checkpoints, relative_path
+    return None, path
+
+  def checkpoint_path(self, checkpoint_id, path):
+    checkpoint_manager, relative_path = self._checkpoint_manager_for_path(path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(path))
+    return checkpoint_manager.checkpoint_path(checkpoint_id, relative_path)
+
+  def checkpoint_blob(self, checkpoint_id, path, create_if_missing=False):
+    checkpoint_manager, relative_path = self._checkpoint_manager_for_path(path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(path))
+    return checkpoint_manager.checkpoint_blob(
+        checkpoint_id, relative_path, create_if_missing=create_if_missing)
+
+  def create_file_checkpoint(self, content, format, path):
+    checkpoint_manager, relative_path = self._checkpoint_manager_for_path(path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(path))
```

Comment:
Why do these all return 500 when the corresponding error modes for the ContentsManager return 403/404?

Reply:
Changed to a 400 error code instead of 500

Reply:
Fixed to return a 400 instead